`LeaderElector` does not work with Kubernetes Mock Server in 6.5.0 because of `Unsupported Media Type`

### Describe the bug

When trying to use the Fabric8 `LeaderElector` for leadership election, it does not seem to work against the Kubernetes Mock Server. It is able to claim the leadership, but it seems not be unable to patch the `Lease` resource due to following exception:

```
2023-03-11 19:53:38 ERROR LeaderElector:152 - Exception occurred while releasing lock 'LeaseLock: test - testlease (f2ed2009-c168-4643-83c9-aa5a17d37da9)'
io.fabric8.kubernetes.client.KubernetesClientException: Failure executing: PATCH at: https://localhost:53856/apis/coordination.k8s.io/v1/namespaces/test/leases/testlease. Message: Unsupported Media Type. Received status: Status(apiVersion=v1, code=415, details=StatusDetails(causes=[], group=null, kind=Unknown, name=null, retryAfterSeconds=null, uid=null, additionalProperties={}), kind=Status, message=Unsupported Media Type, metadata=null, reason=Invalid, status=Failure, additionalProperties={}).
	at io.fabric8.kubernetes.client.KubernetesClientException.copyAsCause(KubernetesClientException.java:238) ~[kubernetes-client-api-6.5.0.jar:?]
	at io.fabric8.kubernetes.client.dsl.internal.OperationSupport.waitForResult(OperationSupport.java:546) ~[kubernetes-client-6.5.0.jar:?]
	at io.fabric8.kubernetes.client.dsl.internal.OperationSupport.handleResponse(OperationSupport.java:566) ~[kubernetes-client-6.5.0.jar:?]
	at io.fabric8.kubernetes.client.dsl.internal.OperationSupport.handlePatch(OperationSupport.java:443) ~[kubernetes-client-6.5.0.jar:?]
	at io.fabric8.kubernetes.client.dsl.internal.OperationSupport.handlePatch(OperationSupport.java:420) ~[kubernetes-client-6.5.0.jar:?]
	at io.fabric8.kubernetes.client.dsl.internal.BaseOperation.handlePatch(BaseOperation.java:717) ~[kubernetes-client-6.5.0.jar:?]
	at io.fabric8.kubernetes.client.dsl.internal.HasMetadataOperation.lambda$patch$2(HasMetadataOperation.java:219) ~[kubernetes-client-6.5.0.jar:?]
	at io.fabric8.kubernetes.client.dsl.internal.HasMetadataOperation.patch(HasMetadataOperation.java:224) ~[kubernetes-client-6.5.0.jar:?]
	at io.fabric8.kubernetes.client.dsl.internal.HasMetadataOperation.patch(HasMetadataOperation.java:239) ~[kubernetes-client-6.5.0.jar:?]
	at io.fabric8.kubernetes.client.dsl.internal.HasMetadataOperation.patch(HasMetadataOperation.java:39) ~[kubernetes-client-6.5.0.jar:?]
	at io.fabric8.kubernetes.client.extension.ResourceAdapter.patch(ResourceAdapter.java:305) ~[kubernetes-client-api-6.5.0.jar:?]
	at io.fabric8.kubernetes.client.extended.leaderelection.resourcelock.ResourceLock.update(ResourceLock.java:65) ~[kubernetes-client-api-6.5.0.jar:?]
	at io.fabric8.kubernetes.client.extended.leaderelection.LeaderElector.release(LeaderElector.java:149) ~[kubernetes-client-api-6.5.0.jar:?]
	at io.fabric8.kubernetes.client.extended.leaderelection.LeaderElector.stopLeading(LeaderElector.java:130) ~[kubernetes-client-api-6.5.0.jar:?]
	at io.fabric8.kubernetes.client.extended.leaderelection.LeaderElector.lambda$null$1(LeaderElector.java:95) ~[kubernetes-client-api-6.5.0.jar:?]
	at java.util.concurrent.CompletableFuture.uniWhenComplete(CompletableFuture.java:863) ~[?:?]
	at java.util.concurrent.CompletableFuture$UniWhenComplete.tryFire(CompletableFuture.java:841) ~[?:?]
	at java.util.concurrent.CompletableFuture.postComplete(CompletableFuture.java:510) ~[?:?]
	at java.util.concurrent.CompletableFuture.complete(CompletableFuture.java:2147) ~[?:?]
	at io.fabric8.kubernetes.client.extended.leaderelection.LeaderElector.lambda$renewWithTimeout$6(LeaderElector.java:179) ~[kubernetes-client-api-6.5.0.jar:?]
	at io.fabric8.kubernetes.client.extended.leaderelection.LeaderElector.lambda$loop$8(LeaderElector.java:266) ~[kubernetes-client-api-6.5.0.jar:?]
	at java.util.concurrent.CompletableFuture$AsyncRun.run(CompletableFuture.java:1804) ~[?:?]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136) ~[?:?]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635) ~[?:?]
	at java.lang.Thread.run(Thread.java:833) ~[?:?]
Caused by: io.fabric8.kubernetes.client.KubernetesClientException: Failure executing: PATCH at: https://localhost:53856/apis/coordination.k8s.io/v1/namespaces/test/leases/testlease. Message: Unsupported Media Type. Received status: Status(apiVersion=v1, code=415, details=StatusDetails(causes=[], group=null, kind=Unknown, name=null, retryAfterSeconds=null, uid=null, additionalProperties={}), kind=Status, message=Unsupported Media Type, metadata=null, reason=Invalid, status=Failure, additionalProperties={}).
	at io.fabric8.kubernetes.client.dsl.internal.OperationSupport.requestFailure(OperationSupport.java:701) ~[kubernetes-client-6.5.0.jar:?]
	at io.fabric8.kubernetes.client.dsl.internal.OperationSupport.requestFailure(OperationSupport.java:681) ~[kubernetes-client-6.5.0.jar:?]
	at io.fabric8.kubernetes.client.dsl.internal.OperationSupport.assertResponseCode(OperationSupport.java:630) ~[kubernetes-client-6.5.0.jar:?]
	at io.fabric8.kubernetes.client.dsl.internal.OperationSupport.lambda$handleResponse$0(OperationSupport.java:591) ~[kubernetes-client-6.5.0.jar:?]
	at java.util.concurrent.CompletableFuture$UniApply.tryFire(CompletableFuture.java:646) ~[?:?]
	at java.util.concurrent.CompletableFuture.postComplete(CompletableFuture.java:510) ~[?:?]
	at java.util.concurrent.CompletableFuture.complete(CompletableFuture.java:2147) ~[?:?]
	at io.fabric8.kubernetes.client.http.StandardHttpClient.lambda$completeOrCancel$5(StandardHttpClient.java:120) ~[kubernetes-client-api-6.5.0.jar:?]
	at java.util.concurrent.CompletableFuture.uniWhenComplete(CompletableFuture.java:863) ~[?:?]
	at java.util.concurrent.CompletableFuture$UniWhenComplete.tryFire(CompletableFuture.java:841) ~[?:?]
	at java.util.concurrent.CompletableFuture.postComplete(CompletableFuture.java:510) ~[?:?]
	at java.util.concurrent.CompletableFuture.complete(CompletableFuture.java:2147) ~[?:?]
	at io.fabric8.kubernetes.client.http.StandardHttpClient.lambda$completeOrCancel$5(StandardHttpClient.java:120) ~[kubernetes-client-api-6.5.0.jar:?]
	at io.fabric8.kubernetes.client.http.StandardHttpClient.lambda$retryWithExponentialBackoff$7(StandardHttpClient.java:159) ~[kubernetes-client-api-6.5.0.jar:?]
	at java.util.concurrent.CompletableFuture.uniWhenComplete(CompletableFuture.java:863) ~[?:?]
	at java.util.concurrent.CompletableFuture$UniWhenComplete.tryFire(CompletableFuture.java:841) ~[?:?]
	at java.util.concurrent.CompletableFuture.postComplete(CompletableFuture.java:510) ~[?:?]
	at java.util.concurrent.CompletableFuture.postFire(CompletableFuture.java:614) ~[?:?]
	at java.util.concurrent.CompletableFuture$UniWhenComplete.tryFire(CompletableFuture.java:844) ~[?:?]
	at java.util.concurrent.CompletableFuture$Completion.exec(CompletableFuture.java:483) ~[?:?]
	at java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:373) ~[?:?]
	at java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1182) ~[?:?]
	at java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1655) ~[?:?]
	at java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1622) ~[?:?]
	at java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:165) ~[?:?]
```

### Fabric8 Kubernetes Client version

6.5.0

### Steps to reproduce

The following reproducer should reproduce the issue. The test does not fail as it does not test anything (it is just used to run against the Mock Server), but you should see the error:

```java
import io.fabric8.kubernetes.client.KubernetesClient;
import io.fabric8.kubernetes.client.extended.leaderelection.LeaderCallbacks;
import io.fabric8.kubernetes.client.extended.leaderelection.LeaderElectionConfigBuilder;
import io.fabric8.kubernetes.client.extended.leaderelection.LeaderElector;
import io.fabric8.kubernetes.client.extended.leaderelection.resourcelock.LeaseLock;
import io.fabric8.kubernetes.client.server.mock.EnableKubernetesMockClient;
import org.junit.jupiter.api.Test;

import java.time.Duration;
import java.util.UUID;

@EnableKubernetesMockClient(crud = true)
public class Fabric8Test {
    private static final String NAME = "testlease";

    // Injected by Fabric8 Mock Kubernetes Server
    @SuppressWarnings("unused")
    private KubernetesClient client;

    @Test
    public void testLeaderElection()   {
        final String lockIdentity = UUID.randomUUID().toString();

        LeaderElector leader = client.leaderElector()
                .withConfig(
                        new LeaderElectionConfigBuilder()
                                .withReleaseOnCancel()
                                .withName("Sample Leader Election configuration")
                                .withLeaseDuration(Duration.ofSeconds(15L))
                                .withLock(new LeaseLock(client.getNamespace(), NAME, lockIdentity))
                                .withRenewDeadline(Duration.ofSeconds(10L))
                                .withRetryPeriod(Duration.ofSeconds(2L))
                                .withLeaderCallbacks(new LeaderCallbacks(
                                        () -> System.out.println("STARTED LEADERSHIP"),
                                        () -> System.out.println("STOPPED LEADERSHIP"),
                                        newLeader -> System.out.printf("New leader elected %s%n", newLeader)))
                                .build())
                .build();
        leader.run();
    }
}
```

### Expected behavior

The `LeaderElector` keeps running and does not fail.

### Runtime

other (please specify in additional context)

### Kubernetes API Server version

other (please specify in additional context)

### Environment

macOS

### Fabric8 Kubernetes Client Logs

```shell
Following logs are from the code shared above:


/Library/Java/JavaVirtualMachines/temurin-17.jdk/Contents/Home/bin/java -ea -Didea.test.cyclic.buffer.size=1048576000 -javaagent:/Applications/IntelliJ IDEA.app/Contents/lib/idea_rt.jar=53852:/Applications/IntelliJ IDEA.app/Contents/bin -Dfile.encoding=UTF-8 -classpath /Users/scholzj/.m2/repository/org/junit/platform/junit-platform-launcher/1.8.2/junit-platform-launcher-1.8.2.jar:/Users/scholzj/.m2/repository/org/junit/vintage/junit-vintage-engine/5.8.2/junit-vintage-engine-5.8.2.jar:/Applications/IntelliJ IDEA.app/Contents/lib/idea_rt.jar:/Applications/IntelliJ IDEA.app/Contents/plugins/junit/lib/junit5-rt.jar:/Applications/IntelliJ IDEA.app/Contents/plugins/junit/lib/junit-rt.jar:/Users/scholzj/development/strimzi/cluster-operator/target/test-classes:/Users/scholzj/development/strimzi/cluster-operator/target/classes:/Users/scholzj/development/strimzi/api/target/classes:/Users/scholzj/.m2/repository/io/fabric8/kubernetes-model-common/6.5.0/kubernetes-model-common-6.5.0.jar:/Users/scholzj/.m2/repository/io/fabric8/kubernetes-model-apiextensions/6.5.0/kubernetes-model-apiextensions-6.5.0.jar:/Users/scholzj/development/strimzi/config-model/target/classes:/Users/scholzj/development/strimzi/operator-common/target/classes:/Users/scholzj/development/strimzi/crd-annotations/target/classes:/Users/scholzj/.m2/repository/org/quartz-scheduler/quartz/2.3.2/quartz-2.3.2.jar:/Users/scholzj/.m2/repository/io/fabric8/openshift-client/6.5.0/openshift-client-6.5.0.jar:/Users/scholzj/.m2/repository/com/github/mifmif/generex/1.0.2/generex-1.0.2.jar:/Users/scholzj/.m2/repository/dk/brics/automaton/automaton/1.11-8/automaton-1.11-8.jar:/Users/scholzj/.m2/repository/io/fabric8/openshift-client-api/6.5.0/openshift-client-api-6.5.0.jar:/Users/scholzj/.m2/repository/io/fabric8/openshift-model-clusterautoscaling/6.5.0/openshift-model-clusterautoscaling-6.5.0.jar:/Users/scholzj/.m2/repository/io/fabric8/openshift-model-operator/6.5.0/openshift-model-operator-6.5.0.jar:/Users/scholzj/.m2/repository/io/fabric8/openshift-model-operatorhub/6.5.0/openshift-model-operatorhub-6.5.0.jar:/Users/scholzj/.m2/repository/io/fabric8/openshift-model-machine/6.5.0/openshift-model-machine-6.5.0.jar:/Users/scholzj/.m2/repository/io/fabric8/openshift-model-whereabouts/6.5.0/openshift-model-whereabouts-6.5.0.jar:/Users/scholzj/.m2/repository/io/fabric8/openshift-model-monitoring/6.5.0/openshift-model-monitoring-6.5.0.jar:/Users/scholzj/.m2/repository/io/fabric8/openshift-model-storageversionmigrator/6.5.0/openshift-model-storageversionmigrator-6.5.0.jar:/Users/scholzj/.m2/repository/io/fabric8/openshift-model-tuned/6.5.0/openshift-model-tuned-6.5.0.jar:/Users/scholzj/.m2/repository/io/fabric8/openshift-model-console/6.5.0/openshift-model-console-6.5.0.jar:/Users/scholzj/.m2/repository/io/fabric8/openshift-model-config/6.5.0/openshift-model-config-6.5.0.jar:/Users/scholzj/.m2/repository/io/fabric8/openshift-model-machineconfig/6.5.0/openshift-model-machineconfig-6.5.0.jar:/Users/scholzj/.m2/repository/io/fabric8/openshift-model-miscellaneous/6.5.0/openshift-model-miscellaneous-6.5.0.jar:/Users/scholzj/.m2/repository/io/fabric8/openshift-model-hive/6.5.0/openshift-model-hive-6.5.0.jar:/Users/scholzj/.m2/repository/io/fabric8/openshift-model-installer/6.5.0/openshift-model-installer-6.5.0.jar:/Users/scholzj/.m2/repository/io/fabric8/kubernetes-client/6.5.0/kubernetes-client-6.5.0.jar:/Users/scholzj/.m2/repository/io/fabric8/kubernetes-httpclient-jdk/6.5.0/kubernetes-httpclient-jdk-6.5.0.jar:/Users/scholzj/.m2/repository/io/fabric8/kubernetes-client-api/6.5.0/kubernetes-client-api-6.5.0.jar:/Users/scholzj/.m2/repository/io/fabric8/kubernetes-model-gatewayapi/6.5.0/kubernetes-model-gatewayapi-6.5.0.jar:/Users/scholzj/.m2/repository/io/fabric8/kubernetes-model-resource/6.5.0/kubernetes-model-resource-6.5.0.jar:/Users/scholzj/.m2/repository/io/fabric8/kubernetes-model-admissionregistration/6.5.0/kubernetes-model-admissionregistration-6.5.0.jar:/Users/scholzj/.m2/repository/io/fabric8/kubernetes-model-autoscaling/6.5.0/kubernetes-model-autoscaling-6.5.0.jar:/Users/scholzj/.m2/repository/io/fabric8/kubernetes-model-batch/6.5.0/kubernetes-model-batch-6.5.0.jar:/Users/scholzj/.m2/repository/io/fabric8/kubernetes-model-certificates/6.5.0/kubernetes-model-certificates-6.5.0.jar:/Users/scholzj/.m2/repository/io/fabric8/kubernetes-model-discovery/6.5.0/kubernetes-model-discovery-6.5.0.jar:/Users/scholzj/.m2/repository/io/fabric8/kubernetes-model-extensions/6.5.0/kubernetes-model-extensions-6.5.0.jar:/Users/scholzj/.m2/repository/io/fabric8/kubernetes-model-flowcontrol/6.5.0/kubernetes-model-flowcontrol-6.5.0.jar:/Users/scholzj/.m2/repository/io/fabric8/kubernetes-model-metrics/6.5.0/kubernetes-model-metrics-6.5.0.jar:/Users/scholzj/.m2/repository/io/fabric8/kubernetes-model-scheduling/6.5.0/kubernetes-model-scheduling-6.5.0.jar:/Users/scholzj/.m2/repository/io/fabric8/kubernetes-model-node/6.5.0/kubernetes-model-node-6.5.0.jar:/Users/scholzj/.m2/repository/org/slf4j/slf4j-api/1.7.36/slf4j-api-1.7.36.jar:/Users/scholzj/.m2/repository/org/snakeyaml/snakeyaml-engine/2.6/snakeyaml-engine-2.6.jar:/Users/scholzj/.m2/repository/com/fasterxml/jackson/datatype/jackson-datatype-jsr310/2.14.2/jackson-datatype-jsr310-2.14.2.jar:/Users/scholzj/.m2/repository/io/fabric8/kubernetes-model-core/6.5.0/kubernetes-model-core-6.5.0.jar:/Users/scholzj/.m2/repository/io/fabric8/kubernetes-model-events/6.5.0/kubernetes-model-events-6.5.0.jar:/Users/scholzj/.m2/repository/io/fabric8/kubernetes-model-policy/6.5.0/kubernetes-model-policy-6.5.0.jar:/Users/scholzj/.m2/repository/io/fabric8/kubernetes-model-rbac/6.5.0/kubernetes-model-rbac-6.5.0.jar:/Users/scholzj/.m2/repository/io/fabric8/kubernetes-model-apps/6.5.0/kubernetes-model-apps-6.5.0.jar:/Users/scholzj/.m2/repository/io/fabric8/kubernetes-model-storageclass/6.5.0/kubernetes-model-storageclass-6.5.0.jar:/Users/scholzj/.m2/repository/io/fabric8/kubernetes-model-networking/6.5.0/kubernetes-model-networking-6.5.0.jar:/Users/scholzj/.m2/repository/io/fabric8/kubernetes-model-coordination/6.5.0/kubernetes-model-coordination-6.5.0.jar:/Users/scholzj/.m2/repository/io/fabric8/openshift-model/6.5.0/openshift-model-6.5.0.jar:/Users/scholzj/.m2/repository/io/fabric8/zjsonpatch/0.3.0/zjsonpatch-0.3.0.jar:/Users/scholzj/.m2/repository/com/fasterxml/jackson/core/jackson-core/2.14.1/jackson-core-2.14.1.jar:/Users/scholzj/.m2/repository/com/fasterxml/jackson/core/jackson-databind/2.14.1/jackson-databind-2.14.1.jar:/Users/scholzj/.m2/repository/com/fasterxml/jackson/core/jackson-annotations/2.14.1/jackson-annotations-2.14.1.jar:/Users/scholzj/.m2/repository/com/fasterxml/jackson/dataformat/jackson-dataformat-yaml/2.14.1/jackson-dataformat-yaml-2.14.1.jar:/Users/scholzj/.m2/repository/org/yaml/snakeyaml/1.33/snakeyaml-1.33.jar:/Users/scholzj/.m2/repository/io/vertx/vertx-core/4.3.8/vertx-core-4.3.8.jar:/Users/scholzj/.m2/repository/io/netty/netty-common/4.1.87.Final/netty-common-4.1.87.Final.jar:/Users/scholzj/.m2/repository/io/netty/netty-buffer/4.1.87.Final/netty-buffer-4.1.87.Final.jar:/Users/scholzj/.m2/repository/io/netty/netty-handler/4.1.87.Final/netty-handler-4.1.87.Final.jar:/Users/scholzj/.m2/repository/io/netty/netty-handler-proxy/4.1.87.Final/netty-handler-proxy-4.1.87.Final.jar:/Users/scholzj/.m2/repository/io/netty/netty-codec-http/4.1.87.Final/netty-codec-http-4.1.87.Final.jar:/Users/scholzj/.m2/repository/io/netty/netty-codec-http2/4.1.87.Final/netty-codec-http2-4.1.87.Final.jar:/Users/scholzj/.m2/repository/io/netty/netty-resolver/4.1.87.Final/netty-resolver-4.1.87.Final.jar:/Users/scholzj/.m2/repository/io/netty/netty-resolver-dns/4.1.87.Final/netty-resolver-dns-4.1.87.Final.jar:/Users/scholzj/.m2/repository/io/netty/netty-codec-dns/4.1.87.Final/netty-codec-dns-4.1.87.Final.jar:/Users/scholzj/development/strimzi/certificate-manager/target/classes:/Users/scholzj/.m2/repository/org/apache/logging/log4j/log4j-core/2.17.2/log4j-core-2.17.2.jar:/Users/scholzj/.m2/repository/org/apache/logging/log4j/log4j-api/2.17.2/log4j-api-2.17.2.jar:/Users/scholzj/.m2/repository/org/apache/logging/log4j/log4j-slf4j-impl/2.17.2/log4j-slf4j-impl-2.17.2.jar:/Users/scholzj/.m2/repository/org/apache/logging/log4j/log4j-jul/2.17.2/log4j-jul-2.17.2.jar:/Users/scholzj/.m2/repository/io/strimzi/kafka-oauth-server/0.11.0/kafka-oauth-server-0.11.0.jar:/Users/scholzj/.m2/repository/io/strimzi/kafka-oauth-server-plain/0.11.0/kafka-oauth-server-plain-0.11.0.jar:/Users/scholzj/.m2/repository/io/strimzi/kafka-oauth-client/0.11.0/kafka-oauth-client-0.11.0.jar:/Users/scholzj/.m2/repository/io/strimzi/kafka-oauth-common/0.11.0/kafka-oauth-common-0.11.0.jar:/Users/scholzj/.m2/repository/com/nimbusds/nimbus-jose-jwt/9.10/nimbus-jose-jwt-9.10.jar:/Users/scholzj/.m2/repository/com/github/stephenc/jcip/jcip-annotations/1.0-1/jcip-annotations-1.0-1.jar:/Users/scholzj/.m2/repository/com/jayway/jsonpath/json-path/2.6.0/json-path-2.6.0.jar:/Users/scholzj/.m2/repository/net/minidev/json-smart/2.4.7/json-smart-2.4.7.jar:/Users/scholzj/.m2/repository/net/minidev/accessors-smart/2.4.7/accessors-smart-2.4.7.jar:/Users/scholzj/.m2/repository/org/junit/jupiter/junit-jupiter-engine/5.8.2/junit-jupiter-engine-5.8.2.jar:/Users/scholzj/.m2/repository/org/junit/platform/junit-platform-engine/1.8.2/junit-platform-engine-1.8.2.jar:/Users/scholzj/.m2/repository/org/junit/jupiter/junit-jupiter-api/5.8.2/junit-jupiter-api-5.8.2.jar:/Users/scholzj/.m2/repository/org/opentest4j/opentest4j/1.2.0/opentest4j-1.2.0.jar:/Users/scholzj/.m2/repository/org/junit/platform/junit-platform-commons/1.8.2/junit-platform-commons-1.8.2.jar:/Users/scholzj/.m2/repository/org/apiguardian/apiguardian-api/1.1.2/apiguardian-api-1.1.2.jar:/Users/scholzj/.m2/repository/org/junit/jupiter/junit-jupiter-params/5.8.2/junit-jupiter-params-5.8.2.jar:/Users/scholzj/.m2/repository/org/hamcrest/hamcrest/2.2/hamcrest-2.2.jar:/Users/scholzj/.m2/repository/io/vertx/vertx-junit5/4.3.8/vertx-junit5-4.3.8.jar:/Users/scholzj/.m2/repository/org/apache/kafka/connect-file/3.4.0/connect-file-3.4.0.jar:/Users/scholzj/.m2/repository/org/apache/kafka/connect-api/3.4.0/connect-api-3.4.0.jar:/Users/scholzj/.m2/repository/javax/ws/rs/javax.ws.rs-api/2.1.1/javax.ws.rs-api-2.1.1.jar:/Users/scholzj/.m2/repository/io/strimzi/strimzi-test-container/0.103.0/strimzi-test-container-0.103.0.jar:/Users/scholzj/.m2/repository/org/testcontainers/testcontainers/1.17.4/testcontainers-1.17.4.jar:/Users/scholzj/.m2/repository/org/rnorth/duct-tape/duct-tape/1.0.8/duct-tape-1.0.8.jar:/Users/scholzj/.m2/repository/org/jetbrains/annotations/17.0.0/annotations-17.0.0.jar:/Users/scholzj/.m2/repository/com/github/docker-java/docker-java-transport-zerodep/3.2.13/docker-java-transport-zerodep-3.2.13.jar:/Users/scholzj/.m2/repository/com/github/docker-java/docker-java-transport/3.2.13/docker-java-transport-3.2.13.jar:/Users/scholzj/.m2/repository/net/java/dev/jna/jna/5.8.0/jna-5.8.0.jar:/Users/scholzj/.m2/repository/org/testcontainers/toxiproxy/1.17.4/toxiproxy-1.17.4.jar:/Users/scholzj/.m2/repository/eu/rekawek/toxiproxy/toxiproxy-java/2.1.7/toxiproxy-java-2.1.7.jar:/Users/scholzj/.m2/repository/com/google/code/gson/gson/2.9.0/gson-2.9.0.jar:/Users/scholzj/.m2/repository/com/github/docker-java/docker-java-api/3.2.13/docker-java-api-3.2.13.jar:/Users/scholzj/.m2/repository/org/apache/commons/commons-compress/1.21/commons-compress-1.21.jar:/Users/scholzj/.m2/repository/junit/junit/4.13.2/junit-4.13.2.jar:/Users/scholzj/.m2/repository/org/hamcrest/hamcrest-core/2.2/hamcrest-core-2.2.jar:/Users/scholzj/.m2/repository/org/apache/kafka/connect-runtime/3.4.0/connect-runtime-3.4.0.jar:/Users/scholzj/.m2/repository/org/apache/kafka/connect-json/3.4.0/connect-json-3.4.0.jar:/Users/scholzj/.m2/repository/com/fasterxml/jackson/datatype/jackson-datatype-jdk8/2.13.4/jackson-datatype-jdk8-2.13.4.jar:/Users/scholzj/.m2/repository/org/apache/kafka/connect-transforms/3.4.0/connect-transforms-3.4.0.jar:/Users/scholzj/.m2/repository/org/apache/kafka/kafka-tools/3.4.0/kafka-tools-3.4.0.jar:/Users/scholzj/.m2/repository/org/apache/kafka/kafka-server-common/3.4.0/kafka-server-common-3.4.0.jar:/Users/scholzj/.m2/repository/com/yammer/metrics/metrics-core/2.2.0/metrics-core-2.2.0.jar:/Users/scholzj/.m2/repository/org/apache/kafka/kafka-log4j-appender/3.4.0/kafka-log4j-appender-3.4.0.jar:/Users/scholzj/.m2/repository/org/slf4j/slf4j-reload4j/1.7.36/slf4j-reload4j-1.7.36.jar:/Users/scholzj/.m2/repository/net/sourceforge/argparse4j/argparse4j/0.7.0/argparse4j-0.7.0.jar:/Users/scholzj/.m2/repository/ch/qos/reload4j/reload4j/1.2.19/reload4j-1.2.19.jar:/Users/scholzj/.m2/repository/org/bitbucket/b_c/jose4j/0.7.9/jose4j-0.7.9.jar:/Users/scholzj/.m2/repository/com/fasterxml/jackson/jaxrs/jackson-jaxrs-json-provider/2.13.4/jackson-jaxrs-json-provider-2.13.4.jar:/Users/scholzj/.m2/repository/com/fasterxml/jackson/jaxrs/jackson-jaxrs-base/2.13.4/jackson-jaxrs-base-2.13.4.jar:/Users/scholzj/.m2/repository/com/fasterxml/jackson/module/jackson-module-jaxb-annotations/2.14.1/jackson-module-jaxb-annotations-2.14.1.jar:/Users/scholzj/.m2/repository/jakarta/activation/jakarta.activation-api/1.2.2/jakarta.activation-api-1.2.2.jar:/Users/scholzj/.m2/repository/org/glassfish/jersey/containers/jersey-container-servlet/2.34/jersey-container-servlet-2.34.jar:/Users/scholzj/.m2/repository/org/glassfish/jersey/containers/jersey-container-servlet-core/2.34/jersey-container-servlet-core-2.34.jar:/Users/scholzj/.m2/repository/org/glassfish/hk2/external/jakarta.inject/2.6.1/jakarta.inject-2.6.1.jar:/Users/scholzj/.m2/repository/org/glassfish/jersey/core/jersey-common/2.34/jersey-common-2.34.jar:/Users/scholzj/.m2/repository/jakarta/annotation/jakarta.annotation-api/1.3.5/jakarta.annotation-api-1.3.5.jar:/Users/scholzj/.m2/repository/org/glassfish/hk2/osgi-resource-locator/1.0.3/osgi-resource-locator-1.0.3.jar:/Users/scholzj/.m2/repository/org/glassfish/jersey/core/jersey-server/2.34/jersey-server-2.34.jar:/Users/scholzj/.m2/repository/org/glassfish/jersey/core/jersey-client/2.34/jersey-client-2.34.jar:/Users/scholzj/.m2/repository/jakarta/validation/jakarta.validation-api/2.0.2/jakarta.validation-api-2.0.2.jar:/Users/scholzj/.m2/repository/jakarta/ws/rs/jakarta.ws.rs-api/2.1.6/jakarta.ws.rs-api-2.1.6.jar:/Users/scholzj/.m2/repository/org/glassfish/jersey/inject/jersey-hk2/2.34/jersey-hk2-2.34.jar:/Users/scholzj/.m2/repository/org/glassfish/hk2/hk2-locator/2.6.1/hk2-locator-2.6.1.jar:/Users/scholzj/.m2/repository/org/glassfish/hk2/external/aopalliance-repackaged/2.6.1/aopalliance-repackaged-2.6.1.jar:/Users/scholzj/.m2/repository/org/glassfish/hk2/hk2-api/2.6.1/hk2-api-2.6.1.jar:/Users/scholzj/.m2/repository/org/glassfish/hk2/hk2-utils/2.6.1/hk2-utils-2.6.1.jar:/Users/scholzj/.m2/repository/org/javassist/javassist/3.25.0-GA/javassist-3.25.0-GA.jar:/Users/scholzj/.m2/repository/javax/xml/bind/jaxb-api/2.3.0/jaxb-api-2.3.0.jar:/Users/scholzj/.m2/repository/javax/activation/activation/1.1.1/activation-1.1.1.jar:/Users/scholzj/.m2/repository/org/eclipse/jetty/jetty-server/9.4.48.v20220622/jetty-server-9.4.48.v20220622.jar:/Users/scholzj/.m2/repository/org/eclipse/jetty/jetty-http/9.4.48.v20220622/jetty-http-9.4.48.v20220622.jar:/Users/scholzj/.m2/repository/org/eclipse/jetty/jetty-io/9.4.48.v20220622/jetty-io-9.4.48.v20220622.jar:/Users/scholzj/.m2/repository/org/eclipse/jetty/jetty-servlet/9.4.48.v20220622/jetty-servlet-9.4.48.v20220622.jar:/Users/scholzj/.m2/repository/org/eclipse/jetty/jetty-security/9.4.48.v20220622/jetty-security-9.4.48.v20220622.jar:/Users/scholzj/.m2/repository/org/eclipse/jetty/jetty-util-ajax/9.4.48.v20220622/jetty-util-ajax-9.4.48.v20220622.jar:/Users/scholzj/.m2/repository/org/eclipse/jetty/jetty-servlets/9.4.48.v20220622/jetty-servlets-9.4.48.v20220622.jar:/Users/scholzj/.m2/repository/org/eclipse/jetty/jetty-continuation/9.4.48.v20220622/jetty-continuation-9.4.48.v20220622.jar:/Users/scholzj/.m2/repository/org/eclipse/jetty/jetty-util/9.4.48.v20220622/jetty-util-9.4.48.v20220622.jar:/Users/scholzj/.m2/repository/org/eclipse/jetty/jetty-client/9.4.48.v20220622/jetty-client-9.4.48.v20220622.jar:/Users/scholzj/.m2/repository/org/reflections/reflections/0.9.12/reflections-0.9.12.jar:/Users/scholzj/.m2/repository/org/apache/maven/maven-artifact/3.8.4/maven-artifact-3.8.4.jar:/Users/scholzj/.m2/repository/org/codehaus/plexus/plexus-utils/3.3.0/plexus-utils-3.3.0.jar:/Users/scholzj/.m2/repository/io/swagger/core/v3/swagger-annotations/2.2.0/swagger-annotations-2.2.0.jar:/Users/scholzj/.m2/repository/org/mockito/mockito-core/4.4.0/mockito-core-4.4.0.jar:/Users/scholzj/.m2/repository/net/bytebuddy/byte-buddy/1.12.8/byte-buddy-1.12.8.jar:/Users/scholzj/.m2/repository/net/bytebuddy/byte-buddy-agent/1.12.8/byte-buddy-agent-1.12.8.jar:/Users/scholzj/.m2/repository/org/objenesis/objenesis/3.2/objenesis-3.2.jar:/Users/scholzj/development/strimzi/test/target/classes:/Users/scholzj/development/strimzi/mockkube/target/classes:/Users/scholzj/development/strimzi/operator-common/target/test-classes:/Users/scholzj/.m2/repository/com/github/spotbugs/spotbugs-annotations/4.7.2/spotbugs-annotations-4.7.2.jar:/Users/scholzj/.m2/repository/com/google/code/findbugs/jsr305/3.0.2/jsr305-3.0.2.jar:/Users/scholzj/.m2/repository/org/apache/kafka/kafka-clients/3.4.0/kafka-clients-3.4.0.jar:/Users/scholzj/.m2/repository/com/github/luben/zstd-jni/1.5.2-1/zstd-jni-1.5.2-1.jar:/Users/scholzj/.m2/repository/org/lz4/lz4-java/1.8.0/lz4-java-1.8.0.jar:/Users/scholzj/.m2/repository/org/xerial/snappy/snappy-java/1.1.8.4/snappy-java-1.1.8.4.jar:/Users/scholzj/.m2/repository/org/apache/zookeeper/zookeeper/3.6.3/zookeeper-3.6.3.jar:/Users/scholzj/.m2/repository/org/apache/yetus/audience-annotations/0.5.0/audience-annotations-0.5.0.jar:/Users/scholzj/.m2/repository/io/netty/netty-transport-native-epoll/4.1.87.Final/netty-transport-native-epoll-4.1.87.Final.jar:/Users/scholzj/.m2/repository/org/apache/zookeeper/zookeeper-jute/3.6.3/zookeeper-jute-3.6.3.jar:/Users/scholzj/.m2/repository/io/netty/netty-transport/4.1.87.Final/netty-transport-4.1.87.Final.jar:/Users/scholzj/.m2/repository/io/netty/netty-transport-native-epoll/4.1.87.Final/netty-transport-native-epoll-4.1.87.Final-linux-x86_64.jar:/Users/scholzj/.m2/repository/io/netty/netty-transport-native-unix-common/4.1.87.Final/netty-transport-native-unix-common-4.1.87.Final.jar:/Users/scholzj/.m2/repository/io/netty/netty-transport-classes-epoll/4.1.87.Final/netty-transport-classes-epoll-4.1.87.Final.jar:/Users/scholzj/.m2/repository/org/scala-lang/scala-library/2.13.9/scala-library-2.13.9.jar:/Users/scholzj/.m2/repository/io/micrometer/micrometer-core/1.9.5/micrometer-core-1.9.5.jar:/Users/scholzj/.m2/repository/org/hdrhistogram/HdrHistogram/2.1.12/HdrHistogram-2.1.12.jar:/Users/scholzj/.m2/repository/org/latencyutils/LatencyUtils/2.0.3/LatencyUtils-2.0.3.jar:/Users/scholzj/.m2/repository/io/micrometer/micrometer-registry-prometheus/1.9.5/micrometer-registry-prometheus-1.9.5.jar:/Users/scholzj/.m2/repository/io/prometheus/simpleclient_common/0.15.0/simpleclient_common-0.15.0.jar:/Users/scholzj/.m2/repository/io/prometheus/simpleclient/0.15.0/simpleclient-0.15.0.jar:/Users/scholzj/.m2/repository/io/prometheus/simpleclient_tracer_otel/0.15.0/simpleclient_tracer_otel-0.15.0.jar:/Users/scholzj/.m2/repository/io/prometheus/simpleclient_tracer_common/0.15.0/simpleclient_tracer_common-0.15.0.jar:/Users/scholzj/.m2/repository/io/prometheus/simpleclient_tracer_otel_agent/0.15.0/simpleclient_tracer_otel_agent-0.15.0.jar:/Users/scholzj/.m2/repository/io/vertx/vertx-micrometer-metrics/4.3.8/vertx-micrometer-metrics-4.3.8.jar:/Users/scholzj/.m2/repository/org/mock-server/mockserver-netty/5.13.2/mockserver-netty-5.13.2.jar:/Users/scholzj/.m2/repository/io/netty/netty-codec/4.1.87.Final/netty-codec-4.1.87.Final.jar:/Users/scholzj/.m2/repository/io/netty/netty-tcnative-boringssl-static/2.0.51.Final/netty-tcnative-boringssl-static-2.0.51.Final.jar:/Users/scholzj/.m2/repository/io/netty/netty-tcnative-classes/2.0.51.Final/netty-tcnative-classes-2.0.51.Final.jar:/Users/scholzj/.m2/repository/io/netty/netty-tcnative-boringssl-static/2.0.51.Final/netty-tcnative-boringssl-static-2.0.51.Final-linux-x86_64.jar:/Users/scholzj/.m2/repository/io/netty/netty-tcnative-boringssl-static/2.0.51.Final/netty-tcnative-boringssl-static-2.0.51.Final-linux-aarch_64.jar:/Users/scholzj/.m2/repository/io/netty/netty-tcnative-boringssl-static/2.0.51.Final/netty-tcnative-boringssl-static-2.0.51.Final-osx-x86_64.jar:/Users/scholzj/.m2/repository/io/netty/netty-tcnative-boringssl-static/2.0.51.Final/netty-tcnative-boringssl-static-2.0.51.Final-osx-aarch_64.jar:/Users/scholzj/.m2/repository/io/netty/netty-tcnative-boringssl-static/2.0.51.Final/netty-tcnative-boringssl-static-2.0.51.Final-windows-x86_64.jar:/Users/scholzj/.m2/repository/commons-io/commons-io/2.11.0/commons-io-2.11.0.jar:/Users/scholzj/.m2/repository/com/google/guava/guava/31.1-jre/guava-31.1-jre.jar:/Users/scholzj/.m2/repository/com/google/guava/failureaccess/1.0.1/failureaccess-1.0.1.jar:/Users/scholzj/.m2/repository/com/google/guava/listenablefuture/9999.0-empty-to-avoid-conflict-with-guava/listenablefuture-9999.0-empty-to-avoid-conflict-with-guava.jar:/Users/scholzj/.m2/repository/org/checkerframework/checker-qual/3.12.0/checker-qual-3.12.0.jar:/Users/scholzj/.m2/repository/com/google/errorprone/error_prone_annotations/2.11.0/error_prone_annotations-2.11.0.jar:/Users/scholzj/.m2/repository/com/google/j2objc/j2objc-annotations/1.3/j2objc-annotations-1.3.jar:/Users/scholzj/.m2/repository/org/mock-server/mockserver-core/5.13.2/mockserver-core-5.13.2.jar:/Users/scholzj/.m2/repository/com/lmax/disruptor/3.4.4/disruptor-3.4.4.jar:/Users/scholzj/.m2/repository/javax/servlet/javax.servlet-api/3.1.0/javax.servlet-api-3.1.0.jar:/Users/scholzj/.m2/repository/io/netty/netty-codec-socks/4.1.87.Final/netty-codec-socks-4.1.87.Final.jar:/Users/scholzj/.m2/repository/com/jcraft/jzlib/1.1.3/jzlib-1.1.3.jar:/Users/scholzj/.m2/repository/com/fasterxml/uuid/java-uuid-generator/4.0.1/java-uuid-generator-4.0.1.jar:/Users/scholzj/.m2/repository/org/bouncycastle/bcprov-jdk15on/1.70/bcprov-jdk15on-1.70.jar:/Users/scholzj/.m2/repository/org/bouncycastle/bcpkix-jdk15on/1.70/bcpkix-jdk15on-1.70.jar:/Users/scholzj/.m2/repository/org/bouncycastle/bcutil-jdk15on/1.70/bcutil-jdk15on-1.70.jar:/Users/scholzj/.m2/repository/org/apache/velocity/velocity-engine-scripting/2.3/velocity-engine-scripting-2.3.jar:/Users/scholzj/.m2/repository/org/apache/velocity/velocity-engine-core/2.3/velocity-engine-core-2.3.jar:/Users/scholzj/.m2/repository/org/apache/velocity/tools/velocity-tools-generic/3.1/velocity-tools-generic-3.1.jar:/Users/scholzj/.m2/repository/commons-beanutils/commons-beanutils/1.9.4/commons-beanutils-1.9.4.jar:/Users/scholzj/.m2/repository/commons-logging/commons-logging/1.2/commons-logging-1.2.jar:/Users/scholzj/.m2/repository/commons-collections/commons-collections/3.2.2/commons-collections-3.2.2.jar:/Users/scholzj/.m2/repository/org/apache/commons/commons-digester3/3.2/commons-digester3-3.2.jar:/Users/scholzj/.m2/repository/com/github/cliftonlabs/json-simple/3.0.2/json-simple-3.0.2.jar:/Users/scholzj/.m2/repository/com/samskivert/jmustache/1.15/jmustache-1.15.jar:/Users/scholzj/.m2/repository/net/javacrumbs/json-unit/json-unit-core/2.33.0/json-unit-core-2.33.0.jar:/Users/scholzj/.m2/repository/com/networknt/json-schema-validator/1.0.68/json-schema-validator-1.0.68.jar:/Users/scholzj/.m2/repository/com/ethlo/time/itu/1.5.1/itu-1.5.1.jar:/Users/scholzj/.m2/repository/io/swagger/parser/v3/swagger-parser/2.0.32/swagger-parser-2.0.32.jar:/Users/scholzj/.m2/repository/io/swagger/parser/v3/swagger-parser-v2-converter/2.0.32/swagger-parser-v2-converter-2.0.32.jar:/Users/scholzj/.m2/repository/io/swagger/swagger-core/1.6.6/swagger-core-1.6.6.jar:/Users/scholzj/.m2/repository/io/swagger/swagger-models/1.6.6/swagger-models-1.6.6.jar:/Users/scholzj/.m2/repository/io/swagger/swagger-annotations/1.6.6/swagger-annotations-1.6.6.jar:/Users/scholzj/.m2/repository/io/swagger/swagger-parser/1.0.59/swagger-parser-1.0.59.jar:/Users/scholzj/.m2/repository/io/swagger/swagger-compat-spec-parser/1.0.59/swagger-compat-spec-parser-1.0.59.jar:/Users/scholzj/.m2/repository/com/github/java-json-tools/json-schema-validator/2.2.14/json-schema-validator-2.2.14.jar:/Users/scholzj/.m2/repository/com/github/java-json-tools/jackson-coreutils-equivalence/1.0/jackson-coreutils-equivalence-1.0.jar:/Users/scholzj/.m2/repository/com/github/java-json-tools/json-schema-core/1.2.14/json-schema-core-1.2.14.jar:/Users/scholzj/.m2/repository/com/github/java-json-tools/uri-template/0.10/uri-template-0.10.jar:/Users/scholzj/.m2/repository/org/mozilla/rhino/1.7.7.2/rhino-1.7.7.2.jar:/Users/scholzj/.m2/repository/com/sun/mail/mailapi/1.6.2/mailapi-1.6.2.jar:/Users/scholzj/.m2/repository/joda-time/joda-time/2.10.5/joda-time-2.10.5.jar:/Users/scholzj/.m2/repository/com/googlecode/libphonenumber/libphonenumber/8.11.1/libphonenumber-8.11.1.jar:/Users/scholzj/.m2/repository/net/sf/jopt-simple/jopt-simple/5.0.4/jopt-simple-5.0.4.jar:/Users/scholzj/.m2/repository/com/github/java-json-tools/json-patch/1.13/json-patch-1.13.jar:/Users/scholzj/.m2/repository/com/github/java-json-tools/msg-simple/1.2/msg-simple-1.2.jar:/Users/scholzj/.m2/repository/com/github/java-json-tools/btf/1.3/btf-1.3.jar:/Users/scholzj/.m2/repository/com/github/java-json-tools/jackson-coreutils/2.0/jackson-coreutils-2.0.jar:/Users/scholzj/.m2/repository/org/apache/httpcomponents/httpclient/4.5.13/httpclient-4.5.13.jar:/Users/scholzj/.m2/repository/org/apache/httpcomponents/httpcore/4.4.13/httpcore-4.4.13.jar:/Users/scholzj/.m2/repository/io/swagger/core/v3/swagger-models/2.2.0/swagger-models-2.2.0.jar:/Users/scholzj/.m2/repository/io/swagger/parser/v3/swagger-parser-core/2.0.32/swagger-parser-core-2.0.32.jar:/Users/scholzj/.m2/repository/io/swagger/parser/v3/swagger-parser-v3/2.0.32/swagger-parser-v3-2.0.32.jar:/Users/scholzj/.m2/repository/io/swagger/core/v3/swagger-core/2.2.0/swagger-core-2.2.0.jar:/Users/scholzj/.m2/repository/jakarta/xml/bind/jakarta.xml.bind-api/3.0.1/jakarta.xml.bind-api-3.0.1.jar:/Users/scholzj/.m2/repository/com/sun/activation/jakarta.activation/2.0.1/jakarta.activation-2.0.1.jar:/Users/scholzj/.m2/repository/com/sun/xml/bind/jaxb-impl/3.0.2/jaxb-impl-3.0.2.jar:/Users/scholzj/.m2/repository/com/sun/xml/bind/jaxb-core/3.0.2/jaxb-core-3.0.2.jar:/Users/scholzj/.m2/repository/org/xmlunit/xmlunit-core/2.9.0/xmlunit-core-2.9.0.jar:/Users/scholzj/.m2/repository/org/xmlunit/xmlunit-placeholders/2.9.0/xmlunit-placeholders-2.9.0.jar:/Users/scholzj/.m2/repository/org/apache/commons/commons-lang3/3.12.0/commons-lang3-3.12.0.jar:/Users/scholzj/.m2/repository/org/apache/commons/commons-text/1.9/commons-text-1.9.jar:/Users/scholzj/.m2/repository/commons-codec/commons-codec/1.13/commons-codec-1.13.jar:/Users/scholzj/.m2/repository/io/github/classgraph/classgraph/4.8.143/classgraph-4.8.143.jar:/Users/scholzj/.m2/repository/org/mock-server/mockserver-client-java/5.13.2/mockserver-client-java-5.13.2.jar:/Users/scholzj/.m2/repository/io/fabric8/kubernetes-server-mock/6.5.0/kubernetes-server-mock-6.5.0.jar:/Users/scholzj/.m2/repository/io/fabric8/mockwebserver/0.2.2/mockwebserver-0.2.2.jar:/Users/scholzj/.m2/repository/com/squareup/okhttp3/mockwebserver/3.14.7/mockwebserver-3.14.7.jar:/Users/scholzj/.m2/repository/com/squareup/okhttp3/okhttp/3.14.7/okhttp-3.14.7.jar:/Users/scholzj/.m2/repository/com/squareup/okio/okio/1.17.2/okio-1.17.2.jar:/Users/scholzj/.m2/repository/io/fabric8/servicecatalog-client/6.5.0/servicecatalog-client-6.5.0.jar:/Users/scholzj/.m2/repository/io/fabric8/servicecatalog-model/6.5.0/servicecatalog-model-6.5.0.jar com.intellij.rt.junit.JUnitStarter -ideVersion5 -junit5 io.strimzi.operator.cluster.operator.assembly.Fabric8Test,testLeaderElection
SLF4J: Class path contains multiple SLF4J bindings.
SLF4J: Found binding in [jar:file:/Users/scholzj/.m2/repository/org/apache/logging/log4j/log4j-slf4j-impl/2.17.2/log4j-slf4j-impl-2.17.2.jar!/org/slf4j/impl/StaticLoggerBinder.class]
SLF4J: Found binding in [jar:file:/Users/scholzj/.m2/repository/org/slf4j/slf4j-reload4j/1.7.36/slf4j-reload4j-1.7.36.jar!/org/slf4j/impl/StaticLoggerBinder.class]
SLF4J: See http://www.slf4j.org/codes.html#multiple_bindings for an explanation.
SLF4J: Actual binding is of type [org.apache.logging.slf4j.Log4jLoggerFactory]
Mar 11, 2023 7:53:28 PM okhttp3.mockwebserver.MockWebServer$3 execute
INFO: MockWebServer[53855] starting to accept connections
Mar 11, 2023 7:53:28 PM okhttp3.mockwebserver.MockWebServer$3 execute
INFO: MockWebServer[53856] starting to accept connections
Mar 11, 2023 7:53:28 PM okhttp3.mockwebserver.MockWebServer$4 processOneRequest
INFO: MockWebServer[53856] received request: GET /apis/coordination.k8s.io/v1/namespaces/test/leases/testlease HTTP/1.1 and responded: HTTP/1.1 404 Client Error
Mar 11, 2023 7:53:28 PM okhttp3.mockwebserver.MockWebServer$4 processOneRequest
INFO: MockWebServer[53856] received request: POST /apis/coordination.k8s.io/v1/namespaces/test/leases HTTP/1.1 and responded: HTTP/1.1 201 OK
New leader elected f2ed2009-c168-4643-83c9-aa5a17d37da9
STARTED LEADERSHIP
Mar 11, 2023 7:53:28 PM okhttp3.mockwebserver.MockWebServer$4 processOneRequest
INFO: MockWebServer[53856] received request: GET /apis/coordination.k8s.io/v1/namespaces/test/leases/testlease HTTP/1.1 and responded: HTTP/1.1 200 OK
Mar 11, 2023 7:53:28 PM okhttp3.mockwebserver.MockWebServer$4 processOneRequest
INFO: MockWebServer[53856] received request: PATCH /apis/coordination.k8s.io/v1/namespaces/test/leases/testlease HTTP/1.1 and responded: HTTP/1.1 415 Client Error
Mar 11, 2023 7:53:30 PM okhttp3.mockwebserver.MockWebServer$4 processOneRequest
INFO: MockWebServer[53856] received request: GET /apis/coordination.k8s.io/v1/namespaces/test/leases/testlease HTTP/1.1 and responded: HTTP/1.1 200 OK
Mar 11, 2023 7:53:30 PM okhttp3.mockwebserver.MockWebServer$4 processOneRequest
INFO: MockWebServer[53856] received request: PATCH /apis/coordination.k8s.io/v1/namespaces/test/leases/testlease HTTP/1.1 and responded: HTTP/1.1 415 Client Error
Mar 11, 2023 7:53:32 PM okhttp3.mockwebserver.MockWebServer$4 processOneRequest
INFO: MockWebServer[53856] received request: GET /apis/coordination.k8s.io/v1/namespaces/test/leases/testlease HTTP/1.1 and responded: HTTP/1.1 200 OK
Mar 11, 2023 7:53:32 PM okhttp3.mockwebserver.MockWebServer$4 processOneRequest
INFO: MockWebServer[53856] received request: PATCH /apis/coordination.k8s.io/v1/namespaces/test/leases/testlease HTTP/1.1 and responded: HTTP/1.1 415 Client Error
Mar 11, 2023 7:53:34 PM okhttp3.mockwebserver.MockWebServer$4 processOneRequest
INFO: MockWebServer[53856] received request: GET /apis/coordination.k8s.io/v1/namespaces/test/leases/testlease HTTP/1.1 and responded: HTTP/1.1 200 OK
Mar 11, 2023 7:53:34 PM okhttp3.mockwebserver.MockWebServer$4 processOneRequest
INFO: MockWebServer[53856] received request: PATCH /apis/coordination.k8s.io/v1/namespaces/test/leases/testlease HTTP/1.1 and responded: HTTP/1.1 415 Client Error
Mar 11, 2023 7:53:36 PM okhttp3.mockwebserver.MockWebServer$4 processOneRequest
INFO: MockWebServer[53856] received request: GET /apis/coordination.k8s.io/v1/namespaces/test/leases/testlease HTTP/1.1 and responded: HTTP/1.1 200 OK
Mar 11, 2023 7:53:36 PM okhttp3.mockwebserver.MockWebServer$4 processOneRequest
INFO: MockWebServer[53856] received request: PATCH /apis/coordination.k8s.io/v1/namespaces/test/leases/testlease HTTP/1.1 and responded: HTTP/1.1 415 Client Error
Mar 11, 2023 7:53:38 PM okhttp3.mockwebserver.MockWebServer$4 processOneRequest
INFO: MockWebServer[53856] received request: GET /apis/coordination.k8s.io/v1/namespaces/test/leases/testlease HTTP/1.1 and responded: HTTP/1.1 200 OK
Mar 11, 2023 7:53:38 PM okhttp3.mockwebserver.MockWebServer$4 processOneRequest
INFO: MockWebServer[53856] received request: PATCH /apis/coordination.k8s.io/v1/namespaces/test/leases/testlease HTTP/1.1 and responded: HTTP/1.1 415 Client Error
2023-03-11 19:53:38 ERROR LeaderElector:152 - Exception occurred while releasing lock 'LeaseLock: test - testlease (f2ed2009-c168-4643-83c9-aa5a17d37da9)'
io.fabric8.kubernetes.client.KubernetesClientException: Failure executing: PATCH at: https://localhost:53856/apis/coordination.k8s.io/v1/namespaces/test/leases/testlease. Message: Unsupported Media Type. Received status: Status(apiVersion=v1, code=415, details=StatusDetails(causes=[], group=null, kind=Unknown, name=null, retryAfterSeconds=null, uid=null, additionalProperties={}), kind=Status, message=Unsupported Media Type, metadata=null, reason=Invalid, status=Failure, additionalProperties={}).
	at io.fabric8.kubernetes.client.KubernetesClientException.copyAsCause(KubernetesClientException.java:238) ~[kubernetes-client-api-6.5.0.jar:?]
	at io.fabric8.kubernetes.client.dsl.internal.OperationSupport.waitForResult(OperationSupport.java:546) ~[kubernetes-client-6.5.0.jar:?]
	at io.fabric8.kubernetes.client.dsl.internal.OperationSupport.handleResponse(OperationSupport.java:566) ~[kubernetes-client-6.5.0.jar:?]
	at io.fabric8.kubernetes.client.dsl.internal.OperationSupport.handlePatch(OperationSupport.java:443) ~[kubernetes-client-6.5.0.jar:?]
	at io.fabric8.kubernetes.client.dsl.internal.OperationSupport.handlePatch(OperationSupport.java:420) ~[kubernetes-client-6.5.0.jar:?]
	at io.fabric8.kubernetes.client.dsl.internal.BaseOperation.handlePatch(BaseOperation.java:717) ~[kubernetes-client-6.5.0.jar:?]
	at io.fabric8.kubernetes.client.dsl.internal.HasMetadataOperation.lambda$patch$2(HasMetadataOperation.java:219) ~[kubernetes-client-6.5.0.jar:?]
	at io.fabric8.kubernetes.client.dsl.internal.HasMetadataOperation.patch(HasMetadataOperation.java:224) ~[kubernetes-client-6.5.0.jar:?]
	at io.fabric8.kubernetes.client.dsl.internal.HasMetadataOperation.patch(HasMetadataOperation.java:239) ~[kubernetes-client-6.5.0.jar:?]
	at io.fabric8.kubernetes.client.dsl.internal.HasMetadataOperation.patch(HasMetadataOperation.java:39) ~[kubernetes-client-6.5.0.jar:?]
	at io.fabric8.kubernetes.client.extension.ResourceAdapter.patch(ResourceAdapter.java:305) ~[kubernetes-client-api-6.5.0.jar:?]
	at io.fabric8.kubernetes.client.extended.leaderelection.resourcelock.ResourceLock.update(ResourceLock.java:65) ~[kubernetes-client-api-6.5.0.jar:?]
	at io.fabric8.kubernetes.client.extended.leaderelection.LeaderElector.release(LeaderElector.java:149) ~[kubernetes-client-api-6.5.0.jar:?]
	at io.fabric8.kubernetes.client.extended.leaderelection.LeaderElector.stopLeading(LeaderElector.java:130) ~[kubernetes-client-api-6.5.0.jar:?]
	at io.fabric8.kubernetes.client.extended.leaderelection.LeaderElector.lambda$null$1(LeaderElector.java:95) ~[kubernetes-client-api-6.5.0.jar:?]
	at java.util.concurrent.CompletableFuture.uniWhenComplete(CompletableFuture.java:863) ~[?:?]
	at java.util.concurrent.CompletableFuture$UniWhenComplete.tryFire(CompletableFuture.java:841) ~[?:?]
	at java.util.concurrent.CompletableFuture.postComplete(CompletableFuture.java:510) ~[?:?]
	at java.util.concurrent.CompletableFuture.complete(CompletableFuture.java:2147) ~[?:?]
	at io.fabric8.kubernetes.client.extended.leaderelection.LeaderElector.lambda$renewWithTimeout$6(LeaderElector.java:179) ~[kubernetes-client-api-6.5.0.jar:?]
	at io.fabric8.kubernetes.client.extended.leaderelection.LeaderElector.lambda$loop$8(LeaderElector.java:266) ~[kubernetes-client-api-6.5.0.jar:?]
	at java.util.concurrent.CompletableFuture$AsyncRun.run(CompletableFuture.java:1804) ~[?:?]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136) ~[?:?]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635) ~[?:?]
	at java.lang.Thread.run(Thread.java:833) ~[?:?]
Caused by: io.fabric8.kubernetes.client.KubernetesClientException: Failure executing: PATCH at: https://localhost:53856/apis/coordination.k8s.io/v1/namespaces/test/leases/testlease. Message: Unsupported Media Type. Received status: Status(apiVersion=v1, code=415, details=StatusDetails(causes=[], group=null, kind=Unknown, name=null, retryAfterSeconds=null, uid=null, additionalProperties={}), kind=Status, message=Unsupported Media Type, metadata=null, reason=Invalid, status=Failure, additionalProperties={}).
	at io.fabric8.kubernetes.client.dsl.internal.OperationSupport.requestFailure(OperationSupport.java:701) ~[kubernetes-client-6.5.0.jar:?]
	at io.fabric8.kubernetes.client.dsl.internal.OperationSupport.requestFailure(OperationSupport.java:681) ~[kubernetes-client-6.5.0.jar:?]
	at io.fabric8.kubernetes.client.dsl.internal.OperationSupport.assertResponseCode(OperationSupport.java:630) ~[kubernetes-client-6.5.0.jar:?]
	at io.fabric8.kubernetes.client.dsl.internal.OperationSupport.lambda$handleResponse$0(OperationSupport.java:591) ~[kubernetes-client-6.5.0.jar:?]
	at java.util.concurrent.CompletableFuture$UniApply.tryFire(CompletableFuture.java:646) ~[?:?]
	at java.util.concurrent.CompletableFuture.postComplete(CompletableFuture.java:510) ~[?:?]
	at java.util.concurrent.CompletableFuture.complete(CompletableFuture.java:2147) ~[?:?]
	at io.fabric8.kubernetes.client.http.StandardHttpClient.lambda$completeOrCancel$5(StandardHttpClient.java:120) ~[kubernetes-client-api-6.5.0.jar:?]
	at java.util.concurrent.CompletableFuture.uniWhenComplete(CompletableFuture.java:863) ~[?:?]
	at java.util.concurrent.CompletableFuture$UniWhenComplete.tryFire(CompletableFuture.java:841) ~[?:?]
	at java.util.concurrent.CompletableFuture.postComplete(CompletableFuture.java:510) ~[?:?]
	at java.util.concurrent.CompletableFuture.complete(CompletableFuture.java:2147) ~[?:?]
	at io.fabric8.kubernetes.client.http.StandardHttpClient.lambda$completeOrCancel$5(StandardHttpClient.java:120) ~[kubernetes-client-api-6.5.0.jar:?]
	at io.fabric8.kubernetes.client.http.StandardHttpClient.lambda$retryWithExponentialBackoff$7(StandardHttpClient.java:159) ~[kubernetes-client-api-6.5.0.jar:?]
	at java.util.concurrent.CompletableFuture.uniWhenComplete(CompletableFuture.java:863) ~[?:?]
	at java.util.concurrent.CompletableFuture$UniWhenComplete.tryFire(CompletableFuture.java:841) ~[?:?]
	at java.util.concurrent.CompletableFuture.postComplete(CompletableFuture.java:510) ~[?:?]
	at java.util.concurrent.CompletableFuture.postFire(CompletableFuture.java:614) ~[?:?]
	at java.util.concurrent.CompletableFuture$UniWhenComplete.tryFire(CompletableFuture.java:844) ~[?:?]
	at java.util.concurrent.CompletableFuture$Completion.exec(CompletableFuture.java:483) ~[?:?]
	at java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:373) ~[?:?]
	at java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1182) ~[?:?]
	at java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1655) ~[?:?]
	at java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1622) ~[?:?]
	at java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:165) ~[?:?]
STOPPED LEADERSHIP
Mar 11, 2023 7:53:38 PM okhttp3.mockwebserver.MockWebServer$3 acceptConnections
INFO: MockWebServer[53856] done accepting connections: Socket closed

Process finished with exit code 0
```

You can see the exception there as well as the Mock Server returning some `415 Client Error` codes.
```


### Additional context

The same code works fine when run with Fabric8 6.3.1.

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -5,6 +5,7 @@
 #### Bugs
 
 #### Improvements
+* Fix #4477 exposing LeaderElector.release to force an elector to give up the lease
 
 #### Dependency Upgrade
 

--- a/junit/kubernetes-server-mock/src/main/java/io/fabric8/kubernetes/client/server/mock/KubernetesMockServerExtension.java
+++ b/junit/kubernetes-server-mock/src/main/java/io/fabric8/kubernetes/client/server/mock/KubernetesMockServerExtension.java
@@ -19,6 +19,7 @@ package io.fabric8.kubernetes.client.server.mock;
 import io.fabric8.kubernetes.client.Client;
 import io.fabric8.kubernetes.client.KubernetesClient;
 import io.fabric8.kubernetes.client.NamespacedKubernetesClient;
+import io.fabric8.kubernetes.client.utils.Serialization;
 import io.fabric8.mockwebserver.Context;
 import io.fabric8.mockwebserver.ServerRequest;
 import io.fabric8.mockwebserver.ServerResponse;
@@ -98,7 +99,8 @@ public class KubernetesMockServerExtension
     EnableKubernetesMockClient a = testClass.getAnnotation(EnableKubernetesMockClient.class);
     final Map<ServerRequest, Queue<ServerResponse>> responses = new HashMap<>();
     mock = a.crud()
-        ? new KubernetesMockServer(new Context(), new MockWebServer(), responses, new KubernetesMixedDispatcher(responses),
+        ? new KubernetesMockServer(new Context(Serialization.jsonMapper()), new MockWebServer(), responses,
+            new KubernetesMixedDispatcher(responses),
             a.https())
         : new KubernetesMockServer(a.https());
     mock.init();

--- a/kubernetes-client-api/src/main/java/io/fabric8/kubernetes/client/extended/leaderelection/LeaderElector.java
+++ b/kubernetes-client-api/src/main/java/io/fabric8/kubernetes/client/extended/leaderelection/LeaderElector.java
@@ -120,22 +120,22 @@ public class LeaderElector {
     if (current == null || !isLeader(current)) {
       return; // not leading
     }
-    try {
-      // update current from latest
-      current = leaderElectionConfig.getLock().get(kubernetesClient);
-      if (current == null || !isLeader(current)) {
-        return; // lost leadership already
-      }
-      if (leaderElectionConfig.isReleaseOnCancel()) {
-        release(current);
-      }
-    } finally {
-      // called regardless of isReleaseOnCancel
+    if (leaderElectionConfig.isReleaseOnCancel()) {
+      release();
+    } else {
       leaderElectionConfig.getLeaderCallbacks().onStopLeading();
     }
   }
 
-  private void release(LeaderElectionRecord current) {
+  /**
+   * Release the leadership if currently held. If not cancelled, the elector will
+   * continue to try and re-acquire the lock.
+   */
+  public synchronized void release() {
+    LeaderElectionRecord current = leaderElectionConfig.getLock().get(kubernetesClient);
+    if (current == null || !isLeader(current)) {
+      return; // lost leadership already
+    }
     try {
       ZonedDateTime now = now();
       final LeaderElectionRecord newLeaderElectionRecord = new LeaderElectionRecord(
@@ -144,9 +144,9 @@ public class LeaderElector {
           now,
           now,
           current.getLeaderTransitions());
-      newLeaderElectionRecord.setVersion(current.getVersion());
 
       leaderElectionConfig.getLock().update(kubernetesClient, newLeaderElectionRecord);
+      updateObserved(newLeaderElectionRecord);
     } catch (KubernetesClientException e) {
       final String lockDescription = leaderElectionConfig.getLock().describe();
       LOGGER.error("Exception occurred while releasing lock '{}'", lockDescription, e);
@@ -218,8 +218,7 @@ public class LeaderElector {
         isLeader ? oldLeaderElectionRecord.getAcquireTime() : now,
         now,
         oldLeaderElectionRecord.getLeaderTransitions() + (isLeader ? 0 : 1));
-    newLeaderElectionRecord.setVersion(oldLeaderElectionRecord.getVersion());
-    leaderElectionConfig.getLock().update(kubernetesClient, newLeaderElectionRecord);
+    lock.update(kubernetesClient, newLeaderElectionRecord);
     updateObserved(newLeaderElectionRecord);
     return true;
   }
@@ -232,6 +231,8 @@ public class LeaderElector {
       final String newLeader = leaderElectionRecord.getHolderIdentity();
       if (!Objects.equals(newLeader, currentLeader)) {
         LOGGER.debug("Leader changed from {} to {}", currentLeader, newLeader);
+        // this will notify even if the newLeader is null or empty, which is the same behavior as the go client
+        // but does not seem entirely correct
         leaderElectionConfig.getLeaderCallbacks().onNewLeader(newLeader);
         if (Objects.equals(currentLeader, leaderElectionConfig.getLock().identity())) {
           leaderElectionConfig.getLeaderCallbacks().onStopLeading();

--- a/kubernetes-client-api/src/main/java/io/fabric8/kubernetes/client/extended/leaderelection/resourcelock/ConfigMapLock.java
+++ b/kubernetes-client-api/src/main/java/io/fabric8/kubernetes/client/extended/leaderelection/resourcelock/ConfigMapLock.java
@@ -18,6 +18,7 @@ package io.fabric8.kubernetes.client.extended.leaderelection.resourcelock;
 import io.fabric8.kubernetes.api.model.ConfigMap;
 import io.fabric8.kubernetes.api.model.ConfigMapBuilder;
 import io.fabric8.kubernetes.api.model.ObjectMeta;
+import io.fabric8.kubernetes.api.model.ObjectMetaBuilder;
 import io.fabric8.kubernetes.client.KubernetesClientException;
 import io.fabric8.kubernetes.client.utils.Serialization;
 import org.slf4j.Logger;
@@ -58,10 +59,10 @@ public class ConfigMapLock extends ResourceLock<ConfigMap> {
   }
 
   @Override
-  protected ConfigMap toResource(LeaderElectionRecord leaderElectionRecord, ObjectMeta meta) {
-    return new ConfigMapBuilder().withMetadata(meta).editMetadata()
-        .addToAnnotations(LEADER_ELECTION_RECORD_ANNOTATION_KEY, Serialization.asJson(leaderElectionRecord))
-        .endMetadata()
+  protected ConfigMap toResource(LeaderElectionRecord leaderElectionRecord, ObjectMetaBuilder meta) {
+    return new ConfigMapBuilder()
+        .withMetadata(
+            meta.addToAnnotations(LEADER_ELECTION_RECORD_ANNOTATION_KEY, Serialization.asJson(leaderElectionRecord)).build())
         .build();
   }
 

--- a/kubernetes-client-api/src/main/java/io/fabric8/kubernetes/client/extended/leaderelection/resourcelock/LeaderElectionRecord.java
+++ b/kubernetes-client-api/src/main/java/io/fabric8/kubernetes/client/extended/leaderelection/resourcelock/LeaderElectionRecord.java
@@ -17,12 +17,9 @@ package io.fabric8.kubernetes.client.extended.leaderelection.resourcelock;
 
 import com.fasterxml.jackson.annotation.JsonCreator;
 import com.fasterxml.jackson.annotation.JsonFormat;
-import com.fasterxml.jackson.annotation.JsonIgnore;
 import com.fasterxml.jackson.annotation.JsonProperty;
-import lombok.AllArgsConstructor;
 import lombok.Builder;
 
-import java.io.Serializable;
 import java.time.Duration;
 import java.time.ZonedDateTime;
 import java.util.Objects;
@@ -35,7 +32,6 @@ import java.util.Objects;
  * @see <a href=
  *      "https://github.com/kubernetes/client-go/blob/1aa326d7304eba6aedc8c89daad615cc7499d1f7/tools/leaderelection/resourcelock/interface.go">leaderelection/resourcelock/interface.go</a>
  */
-@AllArgsConstructor
 @Builder(toBuilder = true)
 public class LeaderElectionRecord {
 
@@ -46,8 +42,6 @@ public class LeaderElectionRecord {
   @JsonFormat(timezone = "UTC", pattern = "yyyy-MM-dd'T'HH:mm:ss.SSSSSS'Z'")
   private final ZonedDateTime renewTime;
   private final int leaderTransitions;
-  @JsonIgnore
-  private transient Serializable version;
 
   @JsonCreator
   public LeaderElectionRecord(
@@ -82,14 +76,6 @@ public class LeaderElectionRecord {
     return leaderTransitions;
   }
 
-  public Serializable getVersion() {
-    return version;
-  }
-
-  public void setVersion(Serializable version) {
-    this.version = version;
-  }
-
   @Override
   public boolean equals(Object o) {
     if (this == o)
@@ -101,13 +87,12 @@ public class LeaderElectionRecord {
         Objects.equals(holderIdentity, that.holderIdentity) &&
         Objects.equals(leaseDuration, that.leaseDuration) &&
         Objects.equals(acquireTime, that.acquireTime) &&
-        Objects.equals(renewTime, that.renewTime) &&
-        Objects.equals(version, that.version);
+        Objects.equals(renewTime, that.renewTime);
   }
 
   @Override
   public int hashCode() {
-    return Objects.hash(holderIdentity, leaseDuration, acquireTime, renewTime, leaderTransitions, version);
+    return Objects.hash(holderIdentity, leaseDuration, acquireTime, renewTime, leaderTransitions);
   }
 
 }

--- a/kubernetes-client-api/src/main/java/io/fabric8/kubernetes/client/extended/leaderelection/resourcelock/LeaseLock.java
+++ b/kubernetes-client-api/src/main/java/io/fabric8/kubernetes/client/extended/leaderelection/resourcelock/LeaseLock.java
@@ -16,6 +16,7 @@
 package io.fabric8.kubernetes.client.extended.leaderelection.resourcelock;
 
 import io.fabric8.kubernetes.api.model.ObjectMeta;
+import io.fabric8.kubernetes.api.model.ObjectMetaBuilder;
 import io.fabric8.kubernetes.api.model.coordination.v1.Lease;
 import io.fabric8.kubernetes.api.model.coordination.v1.LeaseBuilder;
 
@@ -39,8 +40,8 @@ public class LeaseLock extends ResourceLock<Lease> {
   }
 
   @Override
-  protected Lease toResource(LeaderElectionRecord leaderElectionRecord, ObjectMeta meta) {
-    return new LeaseBuilder().withMetadata(meta)
+  protected Lease toResource(LeaderElectionRecord leaderElectionRecord, ObjectMetaBuilder meta) {
+    return new LeaseBuilder().withMetadata(meta.build())
         .withNewSpec()
         .withHolderIdentity(leaderElectionRecord.getHolderIdentity())
         .withLeaseDurationSeconds((int) leaderElectionRecord.getLeaseDuration().get(ChronoUnit.SECONDS))

--- a/kubernetes-client-api/src/main/java/io/fabric8/kubernetes/client/extended/leaderelection/resourcelock/ResourceLock.java
+++ b/kubernetes-client-api/src/main/java/io/fabric8/kubernetes/client/extended/leaderelection/resourcelock/ResourceLock.java
@@ -23,14 +23,13 @@ import io.fabric8.kubernetes.client.KubernetesClient;
 import io.fabric8.kubernetes.client.dsl.base.PatchContext;
 import io.fabric8.kubernetes.client.dsl.base.PatchType;
 
-import java.io.Serializable;
 import java.util.Objects;
-import java.util.Optional;
 
 public abstract class ResourceLock<T extends HasMetadata> implements Lock {
 
   private final ObjectMeta meta;
   private final String identity;
+  private T resource;
 
   public ResourceLock(String namespace, String name, String identity) {
     this(new ObjectMetaBuilder().withNamespace(namespace).withName(name).build(), identity);
@@ -46,23 +45,24 @@ public abstract class ResourceLock<T extends HasMetadata> implements Lock {
   protected abstract Class<T> getKind();
 
   @Override
-  public LeaderElectionRecord get(KubernetesClient client) {
-    return getResource(client).map(this::toRecordInternal).orElse(null);
-  }
-
-  private Optional<T> getResource(KubernetesClient client) {
-    return Optional.ofNullable(client.resources(getKind()).inNamespace(meta.getNamespace()).withName(meta.getName()).get());
-  }
-
-  @Override
-  public void create(KubernetesClient client, LeaderElectionRecord leaderElectionRecord) {
-    client.resource(toResource(leaderElectionRecord, getObjectMeta(leaderElectionRecord.getVersion()))).create();
+  public synchronized LeaderElectionRecord get(KubernetesClient client) {
+    resource = client.resources(getKind()).inNamespace(meta.getNamespace()).withName(meta.getName()).get();
+    if (resource != null) {
+      return toRecord(resource);
+    }
+    return null;
   }
 
   @Override
-  public void update(KubernetesClient client, LeaderElectionRecord leaderElectionRecord) {
-    client.resource(toResource(leaderElectionRecord, getObjectMeta(leaderElectionRecord.getVersion())))
-        .patch(PatchContext.of(PatchType.STRATEGIC_MERGE));
+  public synchronized void create(KubernetesClient client, LeaderElectionRecord leaderElectionRecord) {
+    resource = client.resource(toResource(leaderElectionRecord, getObjectMeta(null))).create();
+  }
+
+  @Override
+  public synchronized void update(KubernetesClient client, LeaderElectionRecord leaderElectionRecord) {
+    Objects.requireNonNull(resource, "get or create must be called first");
+    client.resource(toResource(leaderElectionRecord, getObjectMeta(resource.getMetadata().getResourceVersion())))
+        .patch(PatchContext.of(PatchType.JSON_MERGE));
   }
 
   /**
@@ -72,18 +72,12 @@ public abstract class ResourceLock<T extends HasMetadata> implements Lock {
    * @param meta not null
    * @return
    */
-  protected abstract T toResource(LeaderElectionRecord leaderElectionRecord, ObjectMeta meta);
-
-  protected LeaderElectionRecord toRecordInternal(T resource) {
-    LeaderElectionRecord result = toRecord(resource);
-    result.setVersion(resource.getMetadata().getResourceVersion());
-    return result;
-  }
+  protected abstract T toResource(LeaderElectionRecord leaderElectionRecord, ObjectMetaBuilder meta);
 
   protected abstract LeaderElectionRecord toRecord(T resource);
 
-  protected ObjectMeta getObjectMeta(Serializable version) {
-    return new ObjectMetaBuilder(meta).withResourceVersion((String) version).build();
+  protected ObjectMetaBuilder getObjectMeta(String version) {
+    return new ObjectMetaBuilder(meta).withResourceVersion((String) version);
   }
 
   /**
@@ -100,6 +94,10 @@ public abstract class ResourceLock<T extends HasMetadata> implements Lock {
   @Override
   public String describe() {
     return String.format("%sLock: %s - %s (%s)", getKind().getSimpleName(), meta.getNamespace(), meta.getName(), identity);
+  }
+
+  void setResource(T resource) {
+    this.resource = resource;
   }
 
 }

--- a/kubernetes-client-api/src/main/java/io/fabric8/kubernetes/client/extended/leaderelection/resourcelock/ResourceLock.java
+++ b/kubernetes-client-api/src/main/java/io/fabric8/kubernetes/client/extended/leaderelection/resourcelock/ResourceLock.java
@@ -77,7 +77,7 @@ public abstract class ResourceLock<T extends HasMetadata> implements Lock {
   protected abstract LeaderElectionRecord toRecord(T resource);
 
   protected ObjectMetaBuilder getObjectMeta(String version) {
-    return new ObjectMetaBuilder(meta).withResourceVersion((String) version);
+    return new ObjectMetaBuilder(meta).withResourceVersion(version);
   }
 
   /**

--- a/kubernetes-client-api/src/test/java/io/fabric8/kubernetes/client/extended/leaderelection/LeaderElectorTest.java
+++ b/kubernetes-client-api/src/test/java/io/fabric8/kubernetes/client/extended/leaderelection/LeaderElectorTest.java
@@ -24,6 +24,7 @@ import io.fabric8.kubernetes.client.utils.Utils;
 import org.awaitility.Awaitility;
 import org.junit.jupiter.api.Test;
 import org.mockito.Answers;
+import org.mockito.Mockito;
 
 import java.time.Duration;
 import java.time.Instant;
@@ -55,8 +56,6 @@ import static org.mockito.Mockito.verify;
 import static org.mockito.Mockito.when;
 
 class LeaderElectorTest {
-
-  final static AtomicReference<LeaderElectionRecord> activeLer = new AtomicReference<>(null);
 
   @Test
   void runShouldAbortAfterRenewDeadlineExpired() throws Exception {
@@ -114,17 +113,14 @@ class LeaderElectorTest {
   @Test
   void shouldReleaseWhenCanceled() throws Exception {
     // Given
-    final LeaderElectionConfig lec = mockLeaderElectionConfiguration();
+    AtomicReference<LeaderElectionRecord> activeLer = new AtomicReference<>();
+    final LeaderElectionConfig lec = mockLeaderElectionConfiguration(activeLer);
     final CountDownLatch signal = new CountDownLatch(1);
     final Lock mockedLock = lec.getLock();
     when(lec.isReleaseOnCancel()).thenReturn(true);
     doAnswer(invocation -> {
       LeaderElectionRecord leaderRecord = invocation.getArgument(1, LeaderElectionRecord.class);
-      if (!activeLer.get().getVersion().equals(leaderRecord.getVersion())) {
-        throw new KubernetesClientException("Wrong");
-      }
-      activeLer.set(leaderRecord.toBuilder()
-          .version(String.valueOf(Integer.parseInt(String.valueOf(leaderRecord.getVersion())) + 1)).build());
+      activeLer.set(leaderRecord);
       signal.countDown();
       return null;
     }).when(mockedLock).update(any(), any());
@@ -138,7 +134,6 @@ class LeaderElectorTest {
     // Then
     Awaitility.await().atMost(10, TimeUnit.SECONDS).until(() -> Utils.isNullOrEmpty(activeLer.get().getHolderIdentity()));
     assertEquals(0, activeLer.get().getLeaderTransitions());
-    assertEquals("3", activeLer.get().getVersion());
 
     // create a new elector, they are no good after a single use
     leaderElector = new LeaderElector(mock(NamespacedKubernetesClient.class), lec, CommonThreadPool.get());
@@ -147,6 +142,44 @@ class LeaderElectorTest {
 
     // there should be a transition
     assertEquals(1, activeLer.get().getLeaderTransitions());
+  }
+
+  @Test
+  void shouldRelease() throws Exception {
+    // Given
+    AtomicReference<LeaderElectionRecord> activeLer = new AtomicReference<>();
+    final LeaderElectionConfig lec = mockLeaderElectionConfiguration(activeLer);
+    final CountDownLatch signal = new CountDownLatch(1);
+    final Lock mockedLock = lec.getLock();
+    doAnswer(invocation -> {
+      LeaderElectionRecord leaderRecord = invocation.getArgument(1, LeaderElectionRecord.class);
+      activeLer.set(leaderRecord);
+      signal.countDown();
+      return null;
+    }).when(mockedLock).update(any(), any());
+
+    // When
+    LeaderElector leaderElector = new LeaderElector(mock(NamespacedKubernetesClient.class), lec, CommonThreadPool.get());
+    CompletableFuture<?> started = leaderElector.start();
+    assertTrue(signal.await(10, TimeUnit.SECONDS));
+
+    Mockito.verify(lec.getLeaderCallbacks(), times(1)).onStartLeading();
+
+    leaderElector.release();
+
+    // ensure that release cause us to stop leading
+    Awaitility.await().atMost(10, TimeUnit.SECONDS).until(() -> {
+      Mockito.verify(lec.getLeaderCallbacks()).onStopLeading();
+      return true;
+    });
+
+    // we haven't stopped, so we'll re-acquire
+    Awaitility.await().atMost(10, TimeUnit.SECONDS).until(() -> {
+      Mockito.verify(lec.getLeaderCallbacks(), times(2)).onStartLeading();
+      return true;
+    });
+
+    started.cancel(true);
   }
 
   @Test
@@ -231,9 +264,7 @@ class LeaderElectorTest {
   void loopCancel() throws Exception {
     // Given
     AtomicInteger count = new AtomicInteger();
-    CompletableFuture<?> cf = loop(completion -> {
-      count.getAndIncrement();
-    }, () -> 10L, CommonThreadPool.get());
+    CompletableFuture<?> cf = loop(completion -> count.getAndIncrement(), () -> 10L, CommonThreadPool.get());
     // When
     Awaitility.await().atMost(1, TimeUnit.SECONDS).until(() -> count.get() >= 1);
 
@@ -281,7 +312,12 @@ class LeaderElectorTest {
     assertTrue(result.toMillis() > 1000L);
   }
 
-  private static LeaderElectionConfig mockLeaderElectionConfiguration() throws Exception {
+  private LeaderElectionConfig mockLeaderElectionConfiguration() throws Exception {
+    return mockLeaderElectionConfiguration(new AtomicReference<>());
+  }
+
+  private LeaderElectionConfig mockLeaderElectionConfiguration(AtomicReference<LeaderElectionRecord> activeLer)
+      throws Exception {
     final LeaderElectionConfig lec = mock(LeaderElectionConfig.class, Answers.RETURNS_DEEP_STUBS);
     when(lec.getLeaseDuration()).thenReturn(Duration.ofSeconds(2L));
     when(lec.getRenewDeadline()).thenReturn(Duration.ofSeconds(1L));
@@ -291,7 +327,7 @@ class LeaderElectorTest {
     when(mockedLock.get(any())).thenReturn(null).thenAnswer(invocation -> activeLer.get());
     doAnswer(invocation -> {
       LeaderElectionRecord leaderRecord = invocation.getArgument(1, LeaderElectionRecord.class);
-      activeLer.set(leaderRecord.toBuilder().version("1").build());
+      activeLer.set(leaderRecord);
       return null;
     }).when(mockedLock).create(any(), any());
     return lec;

--- a/kubernetes-client-api/src/test/java/io/fabric8/kubernetes/client/extended/leaderelection/resourcelock/ConfigMapLockTest.java
+++ b/kubernetes-client-api/src/test/java/io/fabric8/kubernetes/client/extended/leaderelection/resourcelock/ConfigMapLockTest.java
@@ -80,10 +80,9 @@ class ConfigMapLockTest {
         .withResourceVersion("313373").build());
     final ConfigMapLock lock = new ConfigMapLock("namespace", "name", "1337");
     // When
-    final LeaderElectionRecord result = lock.toRecordInternal(cm);
+    final LeaderElectionRecord result = lock.toRecord(cm);
     // Then
     assertNotNull(result);
-    assertEquals("313373", result.getVersion());
     assertEquals("1337", result.getHolderIdentity());
     assertEquals(15, result.getLeaseDuration().getSeconds());
     assertEquals(ZonedDateTime.of(2015, 10, 21, 4, 29, 0, 0, ZoneId.of("UTC")), result.getAcquireTime());
@@ -106,10 +105,11 @@ class ConfigMapLockTest {
     // Given
     final LeaderElectionRecord record = new LeaderElectionRecord(
         "1337", Duration.ofSeconds(1), ZonedDateTime.now(), ZonedDateTime.now(), 0);
-    record.setVersion("313373");
     final ConfigMapLock lock = new ConfigMapLock("namespace", "name", "1337");
+    ConfigMap configMap = lock.toResource(record, lock.getObjectMeta("313373"));
+    lock.setResource(configMap);
     // When
-    lock.update(kc, record);
+    lock.update(kc, record.toBuilder().leaseDuration(Duration.ofSeconds(2)).build());
     // Then
     verify(kc.resource(any(ConfigMap.class))).patch(any(PatchContext.class));
   }

--- a/kubernetes-client-api/src/test/java/io/fabric8/kubernetes/client/extended/leaderelection/resourcelock/LeaseLockTest.java
+++ b/kubernetes-client-api/src/test/java/io/fabric8/kubernetes/client/extended/leaderelection/resourcelock/LeaseLockTest.java
@@ -23,6 +23,7 @@ import io.fabric8.kubernetes.client.dsl.base.PatchContext;
 import org.junit.jupiter.api.BeforeEach;
 import org.junit.jupiter.api.Test;
 import org.junit.jupiter.api.function.Executable;
+import org.mockito.Mockito;
 
 import java.time.Duration;
 import java.time.ZoneId;
@@ -82,10 +83,9 @@ class LeaseLockTest {
     lease.setMetadata(new ObjectMetaBuilder().withResourceVersion("313373").build());
     final LeaseLock lock = new LeaseLock("namespace", "name", "1337");
     // When
-    final LeaderElectionRecord result = lock.toRecordInternal(lease);
+    final LeaderElectionRecord result = lock.toRecord(lease);
     // Then
     assertNotNull(result);
-    assertEquals("313373", result.getVersion());
     assertEquals("1337", result.getHolderIdentity());
     assertEquals(15, result.getLeaseDuration().getSeconds());
     assertEquals(ZonedDateTime.of(2015, 10, 21, 4, 29, 0, 0, ZoneId.of("UTC")), result.getAcquireTime());
@@ -108,13 +108,13 @@ class LeaseLockTest {
     // Given
     final LeaderElectionRecord record = new LeaderElectionRecord(
         "1337", Duration.ofSeconds(1), ZonedDateTime.now(), ZonedDateTime.now(), 0);
-    record.setVersion("313373");
     final LeaseLock lock = new LeaseLock("namespace", "name", "1337");
     Lease lease = lock.toResource(record, lock.getObjectMeta("313373"));
+    lock.setResource(lease);
     // When
-    lock.update(kc, record);
+    lock.update(kc, record.toBuilder().leaseDuration(Duration.ofSeconds(2)).build());
     // Then
-    verify(kc.resource(lease)).patch(any(PatchContext.class));
+    verify(kc.resource(Mockito.any(Lease.class))).patch(any(PatchContext.class));
   }
 
   @Test

--- a/kubernetes-tests/src/test/java/io/fabric8/kubernetes/client/mock/LeaderElectionCrudTest.java
+++ b/kubernetes-tests/src/test/java/io/fabric8/kubernetes/client/mock/LeaderElectionCrudTest.java
@@ -1,0 +1,38 @@
+/**
+ * Copyright (C) 2015 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.fabric8.kubernetes.client.mock;
+
+import io.fabric8.kubernetes.client.KubernetesClient;
+import io.fabric8.kubernetes.client.extended.leaderelection.resourcelock.ConfigMapLock;
+import io.fabric8.kubernetes.client.server.mock.EnableKubernetesMockClient;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+
+import static io.fabric8.kubernetes.client.mock.LeaderElectionTest.testAndAssertSingleLeader;
+
+@EnableKubernetesMockClient(crud = true)
+@DisplayName("LeaderElection runs on Kubernetes Mock Server in CRUD mode")
+class LeaderElectionCrudTest {
+
+  KubernetesClient client;
+
+  @Test
+  void singleLeaderConfigMapLockTest() throws Exception {
+    testAndAssertSingleLeader(client, "lead-config-map-crud",
+        new ConfigMapLock("namespace", "name", "lead-config-map-crud"));
+
+  }
+}

--- a/kubernetes-tests/src/test/java/io/fabric8/kubernetes/client/mock/LeaderElectionTest.java
+++ b/kubernetes-tests/src/test/java/io/fabric8/kubernetes/client/mock/LeaderElectionTest.java
@@ -42,14 +42,14 @@ import static org.junit.jupiter.api.Assertions.assertEquals;
 import static org.junit.jupiter.api.Assertions.assertThrows;
 import static org.junit.jupiter.api.Assertions.assertTrue;
 
-@EnableKubernetesMockClient(crud = true)
-public class LeaderElectionTest {
+@EnableKubernetesMockClient
+class LeaderElectionTest {
 
   KubernetesMockServer server;
   KubernetesClient client;
 
   @Test
-  public void singleLeaderConfigMapLockCreateTest() throws Exception {
+  void singleLeaderConfigMapLockCreateTest() throws Exception {
     // Given
     server.expect()
         .post()
@@ -57,12 +57,12 @@ public class LeaderElectionTest {
         .andReturn(200, null)
         .once();
     // When - Then
-    testAndAssertSingleLeader("lead-config-map",
+    testAndAssertSingleLeader(client, "lead-config-map",
         new ConfigMapLock("namespace", "name", "lead-config-map"));
   }
 
   @Test
-  public void singleLeaderConfigMapLockUpdateTest() throws Exception {
+  void singleLeaderConfigMapLockUpdateTest() throws Exception {
     // Given
     server.expect()
         .get()
@@ -82,17 +82,12 @@ public class LeaderElectionTest {
         .andReturn(200, null)
         .once();
     // When - Then
-    testAndAssertSingleLeader("lead-config-map",
-        new ConfigMapLock("namespace", "name", "lead-config-map"));
-
-    // verify crud mock behavior
-    server.clearExpectations();
-    testAndAssertSingleLeader("lead-config-map",
+    testAndAssertSingleLeader(client, "lead-config-map",
         new ConfigMapLock("namespace", "name", "lead-config-map"));
   }
 
   @Test
-  public void singleLeaderLeaseLockCreateTest() throws Exception {
+  void singleLeaderLeaseLockCreateTest() throws Exception {
     // Given
     server.expect()
         .post()
@@ -100,12 +95,12 @@ public class LeaderElectionTest {
         .andReturn(200, null)
         .once();
     // When - Then
-    testAndAssertSingleLeader("lead-lease",
+    testAndAssertSingleLeader(client, "lead-lease",
         new LeaseLock("namespace", "name", "lead-lease"));
   }
 
   @Test
-  public void singleLeaderLeaseLockUpdateTest() throws Exception {
+  void singleLeaderLeaseLockUpdateTest() throws Exception {
     // Given
     server.expect()
         .get()
@@ -129,11 +124,11 @@ public class LeaderElectionTest {
         .andReturn(200, null)
         .once();
     // When - Then
-    testAndAssertSingleLeader("lead-lease",
+    testAndAssertSingleLeader(client, "lead-lease",
         new LeaseLock("namespace", "name", "lead-lease"));
   }
 
-  private void testAndAssertSingleLeader(String id, Lock lock) throws Exception {
+  static void testAndAssertSingleLeader(KubernetesClient client, String id, Lock lock) throws Exception {
     // Given
     final CountDownLatch leaderLatch = new CountDownLatch(1);
     final AtomicReference<String> newLeaderRecord = new AtomicReference<>();
@@ -161,7 +156,7 @@ public class LeaderElectionTest {
     assertEquals(0, leaderLatch.getCount());
     leaderElectorTask.cancel(true);
     executorService.shutdownNow();
-    executorService.awaitTermination(10, TimeUnit.SECONDS);
+    assertTrue(executorService.awaitTermination(10, TimeUnit.SECONDS));
     assertTrue(stoppedLeading.await(10, TimeUnit.SECONDS));
   }
 }

--- a/kubernetes-tests/src/test/java/io/fabric8/kubernetes/client/mock/LeaderElectionTest.java
+++ b/kubernetes-tests/src/test/java/io/fabric8/kubernetes/client/mock/LeaderElectionTest.java
@@ -42,7 +42,7 @@ import static org.junit.jupiter.api.Assertions.assertEquals;
 import static org.junit.jupiter.api.Assertions.assertThrows;
 import static org.junit.jupiter.api.Assertions.assertTrue;
 
-@EnableKubernetesMockClient
+@EnableKubernetesMockClient(crud = true)
 public class LeaderElectionTest {
 
   KubernetesMockServer server;
@@ -82,6 +82,11 @@ public class LeaderElectionTest {
         .andReturn(200, null)
         .once();
     // When - Then
+    testAndAssertSingleLeader("lead-config-map",
+        new ConfigMapLock("namespace", "name", "lead-config-map"));
+
+    // verify crud mock behavior
+    server.clearExpectations();
     testAndAssertSingleLeader("lead-config-map",
         new ConfigMapLock("namespace", "name", "lead-config-map"));
   }

